src/engine-actions/get-payment-preimage.js

```diff
@@ -27,62 +20,82 @@ const POLLING_INTERVAL = 10000
  * then return the hash.
  *
  * @param  {string} paymentHash - Base64 encoded payment hash
- * @returns {GetPaymentPreimageOutcome}
+ * @returns {Promise<GetPaymentPreimageOutcome>}
  * @throws {Error} If payment is in an unknown status
  */
 async function getPaymentPreimage (paymentHash) {
```

Comment:
This method may be unnecessary with the changes to `translateSwap`, so you should consider if you can consolidate this into the other method (or at least the other file).